src/App.tsx

```diff
@@ -1,14 +1,50 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+/* eslint-disable import/no-extraneous-dependencies */
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import debounce from 'lodash.debounce';
+import { TodoModal } from './components/TodoModal';
+
+const getTodosByStatus = (status: string, todos: Todo[]) => {
+  const preperedTodos = [...todos];
+
+  if (status === 'active') {
+    return preperedTodos.filter(todo => todo.completed === false);
+  }
+
+  if (status === 'completed') {
+    return preperedTodos.filter(todo => todo.completed === true);
+  }
+
+  return preperedTodos;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState('all');
+  const [apliedQuery, setApliedQuery] = useState('');
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState(todos[0]);
```

Comment:
```suggestion
  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
```
set the null as default selectedTodo